gui/app/(tabs)/registration.tsx

```diff
@@ -0,0 +1,205 @@
+        setFormData({ ...formData, profilePicture: uri });
+    };
+
+    const handleSubmit = () => {
+        console.log(formData);
+    };
+
+    return (
+        <View style={styles.container}>
+            {/* <Text style={styles.title}>Registrations</Text> */}
+            <TouchableOpacity style={styles.profilePictureContainer} onPress={handleChoosePhoto }>
+                {formData.profilePicture ? (
+                    <Image style={styles.profilePicture} source={{ uri: formData.profilePicture }} />
+                ) : (
+                    <View style={styles.profilePicturePlaceholder}>
+                        <Text style={styles.profilePicturePlaceholderText}>Upload Profile Picture</Text>
+                    </View>
+                )}
+                <Text style={styles.profilePictureButtonText}>Choose Photo</Text>
+            </TouchableOpacity>
+            <TextInput
+                style={styles.input} 
+                placeholderTextColor="#000000"
+                placeholder="First Name"
+                onChangeText={handleFirstNameChange}
+                value={formData.firstName}
+            />
+            <TextInput
+                style={styles.input} 
+                placeholderTextColor="#000000"
+                placeholder="Last Name"
+                onChangeText={handleLastNameChange}
+                value={formData.lastName}
+            />
+            <TextInput
+                style={styles.input} 
+                placeholderTextColor="#000000"
+                placeholder="Username"
+                onChangeText={handleUserNameChange}
+                value={formData.username}
+            />
+            <TextInput
+                style={styles.input} 
+                placeholderTextColor="#000000"
+                placeholder="Email"
+                onChangeText={handleEmailChange}
+                value={formData.email}
+                keyboardType="email-address"
+                autoCapitalize="none"
+            />
+            <TextInput
+                style={styles.input} 
+                placeholderTextColor="#000000"
+                secureTextEntry={true}
+                placeholder="Password"
+                onChangeText={handlePasswordChange}
+                // value={formData.password}
+            />
+            <TouchableOpacity style={styles.button} onPress={handleSubmit}>
+                <Text style={styles.buttonText}>Register</Text>
+            </TouchableOpacity>
+        </View>
+    );
+};
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        alignItems: 'center',
+        justifyContent: 'center',
+    },
+    title: {
+        fontSize: 24,
+        fontWeight: 'bold',
+        marginBottom: 32,
+    },
+    input: {
+        width: '80%',
+        height: 48,
+        borderWidth: 1,
+        borderColor: '#CCCCCC',
+        borderRadius: 8,
+        paddingHorizontal: 16,
+        marginBottom: 16,
+    },
```

Comment:
For some of these things like `title`, it would be great if we could define this once and have it apply to ALL title elements, so we're not re-defining it on each page. I'd also recommend looking into whatever the CSS framework being used is and integrate with that. If we have lots of CSS declarations here, it will be increasingly difficult to keep everything consistent.

I'd also recommend trying to use something other than pixel definitions, those are fairly brittle and don't transfer across screen sizes well.